src/Metadata/Extractor/YamlResourceExtractor.php

```diff
@@ -235,6 +236,27 @@ private function buildOpenapi(array $resource): bool|OpenApiOperation|null
             unset($resource['openapi'][$key]);
         }
 
+        if (\array_key_exists('parameters', $resource['openapi']) && \is_array($openapiParameters = $resource['openapi']['parameters'])) {
+            $parameters = [];
+            $allowedProperties = array_map(fn (\ReflectionProperty $reflProperty): string => $reflProperty->getName(), (new \ReflectionClass(Parameter::class))->getProperties());
```

Comment:
I'm not sure about this, if a parameter is defined it doesn't need to be mapped as a property, I'd remove this alltogehter (with the line 244)

Reply:
This is needed for line 255 ($parameters[] = new Parameter(...$parameter);) . Indeed if I remove it one of my tests will return this error:

1) ApiPlatform\Metadata\Tests\Extractor\YamlExtractorTest::testOpenApiParameters
Error: Unknown named parameter $foo

/var/www/project/core/src/Metadata/Extractor/YamlResourceExtractor.php:250

where 250 is $parameters[] = new Parameter(...$parameter);

All property exept $extensionProperties is in Parameter.php construct.


I can replace `new Parameter(...$parameter)` by 

```
<?php
new Parameter($parameter['name'], $parameter['in'], $parameter['description'] ?? '', $parameter['required'] ?? false, $parameter['deprecated'] ?? false, $parameter['allowEmptyValue'] ?? false, $parameter['schema'] ?? [], $parameter['style'] ?? null, $parameter['explode'] ?? false, $parameter['allowReserved '] ?? false, $parameter['example'] ?? null, isset($parameter['examples']) ? new \ArrayObject($parameter['examples']) : null, isset($parameter['content']) ? new \ArrayObject($parameter['content']) : null);
```
??

Reply:
Okay I understand, yes please avoid relfection here and prefer using the constructor. I'd even use named arguments:

```
new Parameter(
    name: $parameter['name'], 
    in: $parameter['in'], 
    description: $parameter['description'] ?? '',
    required: $parameter['required'] ?? false,
    deprecated: $parameter['deprecated'] ?? false,
    allowEmptyValue: $parameter['allowEmptyValue'] ?? false,
    schema: $parameter['schema'] ?? [],
    style: $parameter['style'] ?? null,
    explode: $parameter['explode'] ?? false,
    allowReserved: $parameter['allowReserved '] ?? false,
    example: $parameter['example'] ?? null,
    examples: isset($parameter['examples']) ? new \ArrayObject($parameter['examples']) : null,
    content: isset($parameter['content']) ? new \ArrayObject($parameter['content']) : null
);
```

Reply:
ok it's reworked